clang/unittests/Format/FormatTestComments.cpp

```diff
@@ -3191,20 +3198,167 @@ TEST_F(FormatTestComments, DontAlignNamespaceComments) {
+               "} // switch also not aligned\n"
+               "int bar;    // new align\n"
+               "int foobar; // group");
+
+  verifyFormat("switch (foo) {\n"
+               "default: {\n"
+               "  int aLongVariable; // with comment\n"
+               "  int f;             // aligned\n"
+               "} // case not aligned\n"
+               "} // switch also not aligned\n"
+               "int bar;    // new align\n"
+               "int foobar; // group");
+
+  verifyFormat("class C {\n"
+               "  int aLongVariable; // with comment\n"
+               "  int f;             // aligned\n"
+               "}; // not aligned\n"
+               "int bar;    // new align\n"
+               "int foobar; // group");
+
+  verifyFormat("struct S {\n"
+               "  int aLongVariable; // with comment\n"
+               "  int f;             // aligned\n"
+               "}; // not aligned\n"
+               "int bar;    // new align\n"
+               "int foobar; // group");
+
+  verifyFormat("union U {\n"
+               "  int aLongVariable; // with comment\n"
+               "  int f;             // aligned\n"
+               "}; // not aligned\n"
+               "int bar;    // new align\n"
+               "int foobar; // group");
+
+  verifyFormat("enum E {\n"
+               "  aLongVariable, // with comment\n"
+               "  f              // aligned\n"
+               "}; // not aligned\n"
+               "int bar;    // new align\n"
+               "int foobar; // group");
+
+  verifyFormat("void foo() {\n"
+               "  {\n"
+               "    int aLongVariable; // with comment\n"
+               "    int f;             // aligned\n"
+               "  } // not aligned\n"
+               "  int bar;    // new align\n"
+               "  int foobar; // group\n"
+               "}");
+
+  verifyFormat("auto longLambda = [] { // comment\n"
+               "  int aLongVariable;   // with comment\n"
+               "  int f;               // aligned\n"
+               "}; // not aligned\n"
+               "int bar;                             // new align\n"
+               "int foobar;                          // group\n"
+               "auto shortLambda = [] { return 5; }; // aligned");
+
+  verifyFormat("auto longLambdaResult = [] { // comment\n"
+               "  int aLongVariable;         // with comment\n"
+               "  int f;                     // aligned\n"
+               "}(); // not aligned\n"
+               "int bar;                               // new align\n"
+               "int foobar;                            // group\n"
+               "auto shortLambda = [] { return 5; }(); // aligned");
+
+  verifyFormat(
+      "auto longLambdaResult = [](auto I, auto J) { // comment\n"
+      "  int aLongVariable;                         // with comment\n"
+      "  int f;                                     // aligned\n"
+      "}(\"Input\", 5); // not aligned\n"
```

Comment:
Do we want this?

I have to say, I was surprised this worked, because I did not think of the case. And I have no opinion on that.

Reply:
I think we do, and it was covered in https://github.com/llvm/llvm-project/pull/68743#pullrequestreview-1691035564:
> Because we should also cover `}(); //` for lambdas.